CHANGELOG.md

```diff
@@ -0,0 +1,178 @@
+        - Return Type: `double`
+        - Description: Gets the maximum value across sites and additionally passes out an array of which site(s) the maximum value was found. Note that the maximum value may be repeated across multiple sites.
+      - `Min()`;
+        - Return Type: `double`
+        - Description: Calculates the minimum value across sites.
+      - `Min(out int[] siteNumbers)`
+        - Return Type: `double`
+        - Description: Gets the minimum value across sites and additionally passes out an array of which site(s) the minimum value was found. Note that the minimum value may be repeated across multiple sites.
+      - `Mean()`
+        - Return Type: `double`
+        - Description: Calculates the mean value across sites.
+    - New constructors added to `PinSiteData<T>` and `SiteData<T>`
+      - `PinSiteData<T>`
+        - `PinSiteData(Dictionary<string, T> pinData, int[] siteNumbers)`: Initializes a PinSiteData object with pin unique data. Where the specified pin unique data value will be repeated across all sites.
+        - `PinSiteData(string[] pinNames, int[] siteNumbers, T data)`: Initializes a PinSiteData object with the same data value across all pins and sites.
+        - `PinSiteData(string[] pinNames, int[] siteNumbers, T[] perPinData)`: Initializes a PinSiteData object with pin unique data. Where the specified site unique data will be repeated across all sites.
+        - `PinSiteData(int[] siteNumbers, string[] pinNames, T[] perSiteData)`: Initializes a PinSiteData object with site unique data. Where the specified site unique data will be repeated across all pins.
+        - `PinSiteData(string[] pinNames, int[] siteNumbers, T[][] perPinPerSiteData)`: Initializes a PinSiteData object with pin and site unique data. Where the specified data value is unique for each pin and each sites, and the input array is formatted in that order: `[pins][sites]`.
+        - `PinSiteData(int[] siteNumbers, string[] pinNames, T[][] perSitePerPinData)`: Initializes a PinSiteData object with site and pin unique data. Where the specified data value is unique for each site and each pin, and the input array is formatted in that order: `[sites][pins]`.
+      - `SiteData<T>`
+        - `SiteData(int[] siteNumbers, T[] perSiteData)`: Initializes a SiteData object with site unique data.
+    - New `Select` Method added to support advanced element-by-element data transformations (such as changing type) and complex math operations.
+      - `SiteData<T>.Select<TResult>(Func<T, TResult> operation)`
+        - Description: Selects each element, of type `T` within the `SiteData<T>` object to perform an operation on that element and return a new `SiteData` object containing the result of that operation, `TResult`.
+        - Return Type: `SiteData<TResult>`
+        - Example: `SiteData<double> MySiteDataAsDoubleObj = MySiteDataAsIntObj.Select(x => (double)x);`
+      - `PinSiteData<T>.Select<TResult>(Func<T, TResult> operation)`
```

Comment:
Should be "PinSiteData<**TResult**>.Select<TResult>(Func<T, TResult> operation)"

Reply:
Seem my reply to your other comment above: https://github.com/ni/semi-test-library-dotnet/pull/126#discussion_r1795343641